python/graphstorm/model/rgcn_encoder.py

```diff
@@ -364,29 +418,47 @@ def skip_last_selfloop(self):
     def reset_last_selfloop(self):
         self.layers[-1].self_loop = self.last_selfloop
 
-    # TODO(zhengda) refactor this to support edge features.
-    def forward(self, blocks, h):
+    def forward(self, blocks, n_h, e_hs=None):
         """ RGCN encoder forward computation.
 
+        .. versionchanged:: 0.4.0
+            Change inputs into node inputs and edge input in v0.4.0 to support edge featuer
+            in RGCN encoder.
+
         Parameters
         ----------
         blocks: list of DGL MFGs
             Sampled subgraph in the list of DGL message flow graphs (MFGs) format. More
             detailed information about DGL MFG can be found in `DGL Neighbor Sampling
             Overview
             <https://docs.dgl.ai/stochastic_training/neighbor_sampling_overview.html>`_.
-        h: dict of Tensor
+        n_h: dict of Tensor
             Input node features for each node type in the format of {ntype: tensor}.
+        e_hs: list of dict of Tensor
+            Input edge features for each edge type in the format of [{etype: tensor}, ...].
+            The length of e_hs should be equal or greater than the number of gnn layers.
+            Default is an empty list '[]'.
 
         Returns
         ----------
         h: dict of Tensor
             New node embeddings for each node type in the format of {ntype: tensor}.
         """
-        for layer, block in zip(self.layers, blocks):
-            h = layer(block, h)
-        return h
+        if e_hs is None:
+            e_hs = []
+
+        if len(e_hs) > 0:
```

Comment:
The default should be an empty list. If directly set `e_hs = []` in argument, it will trigger a pylint dangerous-default-value error. This implementation is suggested by pylint: https://pylint.pycqa.org/en/latest/user_guide/messages/warning/dangerous-default-value.html.

Reply:
removed.